pennylane/math/single_dispatch.py

```diff
@@ -256,27 +255,9 @@ def _ndim_tf(tensor):
 ar.register_function("tensorflow", "ndim", _ndim_tf)
 
 
-def _take_tf(tensor, indices, axis=None):
-    """Implement a TensorFlow version of np.take"""
+def _take_tf(tensor, indices, axis=None, **kwargs):
     tf = _i("tf")
-
-    if isinstance(indices, numbers.Number):
-        indices = [indices]
-
-    indices = tf.convert_to_tensor(indices)
-
-    if np.any(indices < 0):
-        # Unlike NumPy, TensorFlow doesn't support negative indices.
-        dim_length = tf.size(tensor).numpy() if axis is None else tf.shape(tensor)[axis]
-        indices = tf.where(indices >= 0, indices, indices + dim_length)
-
-    if axis is None:
-        # Unlike NumPy, if axis=None TensorFlow defaults to the first
-        # dimension rather than flattening the array.
-        data = tf.reshape(tensor, [-1])
-        return tf.gather(data, indices)
-
-    return tf.gather(tensor, indices, axis=axis)
+    return tf.experimental.numpy.take(tensor, indices, axis=axis, **kwargs)

```

Comment:
was our old implementation broken in some way, or is this just a reduction of code to maintain?

Reply:
Old code left a singleton dimension. All other frameworks squeezed out the singleton dimension except for tensorflow, and `tf.experimental.numpy.take` squeezes out the singleton dimension.